sled-agent/src/storage/dump_setup.rs

```diff
@@ -0,0 +1,153 @@
+use crate::storage_manager::DiskWrapper;
+use camino::Utf8PathBuf;
+use illumos_utils::zpool::ZpoolHealth;
+use omicron_common::disk::DiskIdentity;
+use sled_hardware::DiskVariant;
+use slog::Logger;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::MutexGuard;
+
+#[derive(Default)]
+pub struct DumpSetup {
+    savecore_lock: Arc<std::sync::Mutex<()>>,
```

Comment:
Since this lock is particularly opt-in, can we document what it's doing? Looks like "preventing us from running savecore concurrently", right?